distributed/utils.py

```diff
@@ -1013,6 +1013,26 @@ def ensure_bytes(s):
             ) from e
 
 
+def ensure_memoryview(obj):
+    """Ensure `obj` is a `memoryview` that is 1-D contiguous of `uint8` type"""
+    mv: memoryview
+    if type(obj) is memoryview:
+        mv = obj
+    else:
+        mv = memoryview(obj)
+
+    if not mv.nbytes:
+        # empty
+        return memoryview(b"")
```

Comment:
Why make a new one here if it's the same as what we already have?

Reply:
It may not be. For example here's an N-D `memoryview` of a different format.

<details>

```python
In [1]: import numpy as np

In [2]: a = np.arange(12).reshape(3, 4)[:0, ::2]

In [3]: mv = a.data

In [4]: mv.nbytes
Out[4]: 0

In [5]: mv.shape
Out[5]: (0, 2)

In [6]: mv.strides
Out[6]: (16, 8)

In [7]: mv.format
Out[7]: 'l'
```

</details>

A second reason to make a new one is we don't actually need the memory backed by the original `memoryview` in this case. So this allows us to free that memory (assuming nothing else is holding it).

Reply:
ok